ext/liquid_c/vm.c

```diff
@@ -54,9 +51,9 @@ void liquid_vm_render(block_body_t *body, VALUE context, VALUE output)
                 rb_bug("invalid opcode: %u", ip[-1]);
         }
         if (RB_UNLIKELY(is_captured)) {
-            rb_funcall(resource_limits, id_increment_write_score, 1, output);
+            resource_limits_increment_write_score(resource_limits, output);
         } else if (RSTRING_LEN(output) > render_length_limit) {
-            rb_funcall(resource_limits, id_raise_limits_reached, 0);
+            resource_limits_raise_limits_reached(resource_limits);
         }
```

Comment:
Semantically, this is `increment_write_score`, just written like this as an optimization.  I extracted it to an inline `increment_write_score` function in https://github.com/Shopify/liquid-c/pull/59 which we could probably move to resource_limits.h and just have a pointer to the resource_limits struct and the output string as parameters.

I don't think we need the `is_captured` or `render_length_limit` local variables in this function now, they were just here to reduce ruby function calls.